lib/clientBuilder.js

```diff
@@ -159,7 +159,23 @@ proto.features.isTokenVerifySupported = function() {
 
 // { username, password, (relayState), (context) }
 proto.signIn = function (opts) {
-  return tx.postToTransaction(this, '/api/v1/authn', opts);
+  var sdk = this;
+  var body = opts || {};
+  return new Q(!body.sendFingerprint ? {} :
+    sdk.fingerprint()
+    .then(function(fingerprint) {
+      return {
```

Comment:
another option would be to just add it to ```opts``` directly, and that way you don't need to create ```body``` nor add it to ```tx.postToTransacton```... just an idea to save a variable, but either works
```
opts['headers']['X-Device-Fingerprint'] = fingerprint;
```

Reply:
I could remove the `body` variable assignment above, but I'll still need to do it below to delete the `sendFingerprint` property without affecting the `opts` object they passed in.

```javascript
var body = util.clone(opts || {});
delete body.sendFingerprint;
```

Reply:
If we're really opposed to the variable declaration, it could be:

```javascript
opts = util.clone(opts || {});
delete opts.sendFingerprint;
```

Reply:
not opposed, was just throwing an idea... but you are right, changing ```opts``` directly is not a good practice, much better to just pass the extra argument to ```http.post``` and let it deal with merging the options. So all good 👍